tsfile/src/main/java/org/apache/iotdb/tsfile/read/query/timegenerator/node/LeafNode.java

```diff
@@ -20,44 +20,28 @@
 
 import java.io.IOException;
 import org.apache.iotdb.tsfile.read.common.BatchData;
+import org.apache.iotdb.tsfile.read.common.TimeSeries;
 import org.apache.iotdb.tsfile.read.reader.series.AbstractFileSeriesReader;
 
 public class LeafNode implements Node {
 
   private AbstractFileSeriesReader reader;
 
-  private BatchData data = null;
-
-  private boolean gotData = false;
+  private BatchData cacheData;
 
   public LeafNode(AbstractFileSeriesReader reader) {
     this.reader = reader;
   }
 
   @Override
   public boolean hasNext() throws IOException {
-
-    if (gotData) {
-      data.next();
-      gotData = false;
-    }
-
-    if (data == null || !data.hasCurrent()) {
-      if (reader.hasNextBatch()) {
-        data = reader.nextBatch();
-      } else {
-        return false;
-      }
-    }
-
-    return data.hasCurrent();
+    return reader.hasNextBatch();
   }
 
   @Override
-  public long next() {
-    long time = data.currentTime();
-    gotData = true;
-    return time;
+  public TimeSeries next() throws IOException {
```

Comment:
```suggestion
  public TimeSeries nextTimes() throws IOException {
```

Reply:
that `Node` Interface was rename to `nextTimeColumn`